src/swap/utils.ts

```diff
@@ -34,6 +35,30 @@ export function getQuote(params: {
   return getFixedOutputSwapQuote(params);
 }
 
+function getQuotePromiseWrapper(promises: Promise<SwapQuoteWithPool>[]) {
+  return Promise.allSettled(promises).then((results) => {
+    if (
+      results.every(
+        (result) =>
+          result.status === "rejected" && result.reason instanceof SwapQuoteError
```

Comment:
TODO: Check error type instead of `instanceof` check

Reply:
I renamed SwapQuoteError as InsuffucientLiquidityError, since it is only responsible for liquidity, I think I can keep instanceof check

Reply:
yes it should be fine